### PR TITLE
add date input to customize publish date value

### PR DESCRIPTION
**Pull request by fuentesloic**
following #311

### CODE REVIEW

File: blog/views.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ def article_edit(id):
         article.title = request.form.get('title')
         article.content = request.form.get('content')
         if bool(request.form.get('published')):
```
Comment (vinyll) [inline blog/views.py]: Editing the publication date is probably not the best idea of the world.
The issues says we should, but the need is to preserve the initial publication date.
I would therefore only set a publication date when it has none yet.
Therefore, an article that has a publication date should never be able to change it; it's it initial publication date:

```python
if bool(request.form.get('published') and not  article.publication_date):
    article.publication_date = datetime.datetime.utcnow()
```

However, doing so, I'm not sure we could still unpublish an article 🤔

File: blog/views.py [hunk 2]
```diff
@@ -67,10 +67,8 @@ def article_edit(id):
     if request.method == 'POST':
         article.title = request.form.get('title')
         article.content = request.form.get('content')
-        if bool(request.form.get('published')):
+        if bool(request.form.get('published') and not article.publication_date):
```
Comment (vinyll) [inline blog/views.py]: We actually need to convert to boolean the `request.form.get('published')` part:

```python
if bool(request.form.get('published')) and not article.publication_date:
```

**Comment by fuentesloic** [comment]
@vinyll Indeed doing this, we can't unpublished an article when it's have been published once.

Could we add an `booleanField` in article model to deal with ? not to hard after with article in production ?

File: blog/views.py [hunk 3]
```diff
@@ -67,10 +69,8 @@ def article_edit(id):
     if request.method == 'POST':
         article.title = request.form.get('title')
         article.content = request.form.get('content')
-        if bool(request.form.get('published')):
+        if bool(request.form.get('published')) and not article.publication_date:
             article.publication_date = datetime.datetime.utcnow()
```
Comment (vinyll) [inline blog/views.py:74]: If you create an `article.published` property, you need to use it for publishing / unpublishing:

```python
article.published = bool(request.form.get('published'))
if article.published and not article.publication_date:
    article.publication_date = datetime.datetime.utcnow()
```

**Review by vinyll — CHANGES_REQUESTED** [review]

File: blog/views.py (hunk 3 above)
Comment (fuentesloic) [reply blog/views.py:74]: I have to add in the model right ?

Comment (vinyll) [reply blog/views.py:74]: You're totally right my python friend. 🐍

**Comment by fuentesloic** [comment]
@vinyll and now ??

File: blog/test.py [hunk 4]
```diff
@@ -174,16 +175,26 @@ def test_language_detection(self):
...
 
     def test_unpublished_article_does_not_appear_in_articles(self):
         Article(title='title', content='<p>content</p>',
+                published=False,
                 publication_date=None,
                 author=self.user).save()
         result = self.client.get('/articles/')
         self.assertNotIn(b'<article class=preview>', result.data)
+        
+    def test_published_then_unpublished_article_does_not_appear_in_articles(self):
```
Comment (vinyll) [inline blog/test.py]: the test does not publish then unpublish.
You may mean `test_unpublished_article_does_not_appear_in_articles` ?

**Review by vinyll — APPROVED** [review]

**Pull request merged**